pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -1487,6 +1498,46 @@ static void updateColumnNames(String rawTableName, PinotQuery pinotQuery, boolea
     }
   }
 
+  private static void expandStarExpressionToActualColumns(PinotQuery pinotQuery, Map<String, String> columnNameMap,
+      Expression selectStarExpr) {
+    List<Expression> originalSelections = pinotQuery.getSelectList();
+    // Avoid using stream apis in query path because we have found that it has poorer performance compared to
+    // regular apis.
+    Set<String> originallySelectedColumnNames = new HashSet<>();
+    for (Expression originalSelection : originalSelections) {
+      if (originalSelection.isSetIdentifier()) {
+        originallySelectedColumnNames.add(originalSelection.getIdentifier().getName());
+      }
+    }
+    List<Expression> newSelections = new ArrayList<>();
+    for (Expression selection : originalSelections) {
+      if (selection.equals(selectStarExpr)) {
+        //expand '*' to actual columns, exclude default virtual columns
+        for (String tableCol : columnNameMap.values()) {
+          //we exclude default virtual columns and those columns that are already a part of originalSelections (to
+          // dedup columns that are selected multiple times)
```

Comment:
I thought we decided not to dedup since that is the behavior that mysql and postgres follow?

Reply:
The output remains cleaner if we dedup. I like @siddharthteotia's idea that if the user wants to the same column multiple times, he should use aliasing (which the change handles). @Jackie-Jiang What are your thoughts on this?

Reply:
So by extension, `SELECT *, * FROM baseballStats` should also be deduped right?

Reply:
Also from what I can see queries like `SELECT playerID, playerID FROM baseballStats` are currently supported in Pinot without the deduce. So whatever scheme we pick should be consistent otherwise it will cause confusion later and be difficult to change.

Reply:
I feel there is no SQL standard for this. Returning duplicate columns can add extra workload and traffic without providing much value to the user, so I prefer deduping the columns.

Reply:
@Jackie-Jiang mysql doesn't dudup in any of the cases listed above, except that multiple stars are not allowed in the select list. I don't have access to postgresql, but I believe (as @suddendust tested earlier) the behavior is the same as mysql except that postgresql allows multiple stars in the select list.

The w3cschools console might be custom UI logic (note how they change column name for query `SELECT CustomerID,  FROM Customers` without the use of aliasing) . Nothing wrong with deduding per se and either approach would work, but I think we need to maintain consistency. For example:
* if we are deduping select list, then shouldn't `select *, *` be also deduped (w3c sql console is doing dudupe here)?
*  if we decide to dedup `select *, *` then shouldn't `select playerID, playerID` (existing functionality which is not being currently deduped) also be deduced?
* what about `select playerID, *, playerID` - how will dudupe work in this case? (since we currently don't dedupe `playerID, playerID` but will dedup `playerID, *`)
*  I am wondering if we have an example of an actual commercial database that is doing dedup of select list?

Reply:
I tried mysql, postgres and sql server, seems all of them don't dedup the columns automatically, so probably we should follow this convention